frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -310,7 +360,8 @@ end
 
 function ReaderBookmark:onSaveSettings()
     self.ui.doc_settings:saveSetting("bookmarks", self.bookmarks)
-    self.ui.doc_settings:makeTrue("bookmarks_sorted")
+    self.ui.doc_settings:delSetting("bookmarks_sorted")
+    self.ui.doc_settings:makeTrue("bookmarks_sorted_20220106")
```

Comment:
I would let `self.ui.doc_settings:makeTrue("bookmarks_sorted")` and don't mind it.
We don't really need to have docsettings that tidy :)
But more importantly, think about people trying an older KOReader version (for comparing behaviour, regressions...):
- they have a document with `bookmarks_sorted`, upgrade, get `bookmarks_sorted_20220106`, and get `bookmarks_sorted` removed
- they switch back to v2021.10: the old fixBookmark is run (because `bookmarks_sorted` is not there) which might re-order your nice ordering.
(I don't mind much how bookmark behave on v2021.10, we should just try to be sure it doesn't crash)
- they switch back to v2022.02: you don't run your nice re-ordering because `bookmarks_sorted_20220106` is there, but you have badly re-ordered bookmarks as done by v2021.10 (and you get reports for bugs and issues you have a hard time understanding :)